### ISSUE TITLE
'package:flutter/src/rendering/shifted_box.dart': Failed assertion: line 55 pos 14: '!debugNeedsLayout': is not true.

### ISSUE THREAD

**Issue by kecson** [flutter/flutter#36430]
## Steps to Reproduce

1. in the listview, return a ListTile:
```dart
ListTile(
                        title: Row(
                          children: <Widget>[
                            _getDeviceIcon(device),
                            const SizedBox(width: 8),
                            Expanded(
                                flex: 3,
                                child: Text(device.name,
                                    style: textStyle,
                                    overflow: TextOverflow.ellipsis)),
                            Expanded(
                                flex: 2,
                                child: Text(device.devType,
                                    textAlign: TextAlign.center,
                                    style: textStyle,
                                    overflow: TextOverflow.ellipsis)),
                            Text(errorCount > 0 ? '异常' : '正常',
                                style: Theme.of(context)
                                    .textTheme
                                    .caption
                                    .copyWith(
                                        color: errorCount > 0
                                            ? Colors.redAccent
                                            : Theme.of(context)
                                                .textTheme
                                                .caption
                                                .color)),
                          ],
                        ),
                        trailing: SizedBox(width: 0, height: 0),
                        subtitle: _buildDeviceAlarmInfoWidget(device, context),
                        onTap: () => _onClickDevice(context, device, index));
```

## Logs

```
 I/flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
 I/flutter: The following assertion was thrown during performLayout():
 I/flutter: 'package:flutter/src/rendering/shifted_box.dart': Failed assertion: line 55 pos 14:
 I/flutter: '!debugNeedsLayout': is not true.
 I/flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
 I/flutter: more information in this error message to help you determine and fix the underlying cause.
 I/flutter: In either case, please report this assertion by filing a bug on GitHub:
 I/flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
 I/flutter: When the exception was thrown, this was the stack:
 I/flutter: #2      RenderShiftedBox.computeDistanceToActualBaseline (package:flutter/src/rendering/shifted_box.dart:55:14)
 I/flutter: #3      RenderBox.getDistanceToActualBaseline.<anonymous closure> (package:flutter/src/rendering/box.dart:1875:50)
 I/flutter: #4      __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:281:23)
 I/flutter: #5      RenderBox.getDistanceToActualBaseline (package:flutter/src/rendering/box.dart:1875:22)
 I/flutter: #6      _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.computeDistanceToActualBaseline (package:flutter/src/rendering/proxy_box.dart:98:20)
 I/flutter: #7      RenderBox.getDistanceToActualBaseline.<anonymous closure> (package:flutter/src/rendering/box.dart:1875:50)
 I/flutter: #8      __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:281:23)
 I/flutter: #9      RenderBox.getDistanceToActualBaseline (package:flutter/src/rendering/box.dart:1875:22)
 I/flutter: #10     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.computeDistanceToActualBaseline (package:flutter/src/rendering/proxy_box.dart:98:20)
 I/flutter: #11     RenderBox.getDistanceToActualBaseline.<anonymous closure> (package:flutter/src/rendering/box.dart:1875:50)
 I/flutter: #12     __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:281:23)
 I/flutter: #13     RenderBox.getDistanceToActualBaseline (package:flutter/src/rendering/box.dart:1875:22)
 I/flutter: #14     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.computeDistanceToActualBaseline (package:flutter/src/rendering/proxy_box.dart:98:20)
 I/flutter: #15     RenderBox.getDistanceToActualBaseline.<anonymous closure> (package:flutter/src/rendering/box.dart:1875:50)
 I/flutter: #16     __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:281:23)
 I/flutter: #17     RenderBox.getDistanceToActualBaseline (package:flutter/src/rendering/box.dart:1875:22)
 I/flutter: #18     _RenderFlex&RenderBox&ContainerRenderObjectMixin&RenderBoxContainerDefaultsMixin.defaultComputeDistanceToHighestActualBaseline (package:flutter/src/rendering/box.dar
 I/flutter: #19     RenderFlex.computeDistanceToActualBaseline (package:flutter/src/rendering/flex.dart:602:14)
 I/flutter: #20     RenderBox.getDistanceToActualBaseline.<anonymous closure> (package:flutter/src/rendering/box.dart:1875:50)
 I/flutter: #21     __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:281:23)
 I/flutter: #22     RenderBox.getDistanceToActualBaseline (package:flutter/src/rendering/box.dart:1875:22)
 I/flutter: #23     RenderBox.getDistanceToBaseline (package:flutter/src/rendering/box.dart:1858:27)
 I/flutter: #24     _RenderListTile._boxBaseline (package:flutter/src/material/list_tile.dart:1294:16)
 I/flutter: #25     _RenderListTile.performLayout (package:flutter/src/material/list_tile.dart:1373:32)
 I/flutter: #26     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #27     RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:206:11)
 I/flutter: #28     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #29     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
 I/flutter: #30     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #31     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
 I/flutter: #32     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #33     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
 I/flutter: #34     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #35     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
 I/flutter: #36     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #37     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
 I/flutter: #38     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #39     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
 I/flutter: #40     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #41     RenderSliverList.performLayout (package:flutter/src/rendering/sliver_list.dart:165:27)
 I/flutter: #42     RenderObject.layout (package:flutter/src/rendering/object.dart:1619:7)
 I/flutter: #43     RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:406:13)
 I/flutter: #44     RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1334:12)
 I/flutter: #45     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1252:20)
 I/flutter: #46     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1496:7)
 I/flutter: #47     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:765:18)
 I/flutter: #48     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/re
 I/flutter: #49     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package
 I/flutter: #50     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (p
 I/flutter: #51     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1016:15)
 I/flutter: #52     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:958:9)
 I/flutter: #53     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:874:5)
 I/flutter: #57     _invoke (dart:ui/hooks.dart:236:10)
 I/flutter: #58     _drawFrame (dart:ui/hooks.dart:194:3)
 I/flutter: (elided 5 frames from class _AssertionError and package dart:async)
 I/flutter: The following RenderObject was being processed when the exception was fired: _RenderListTile#ad779 relayoutBoundary=up9 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDAT
 I/flutter:   creator: _ListTile ← MediaQuery ← Padding ← SafeArea ← Semantics ← Listener ← _GestureSemantics ←
 I/flutter:     RawGestureDetector ← GestureDetector ← Listener ← InkWell ← ListTile ← ⋯
 I/flutter:   parentData: offset=Offset(16.0, 0.0) (can use size)
 I/flutter:   constraints: BoxConstraints(w=328.0, 0.0<=h<=Infinity)
 I/flutter:   size: Size(328.0, 56.0)
 I/flutter: This RenderObject had the following descendants (showing up to depth 5):
 I/flutter:     title: RenderFlex#38cdd relayoutBoundary=up10 NEEDS-PAINT
 I/flutter:       child 1: RenderSemanticsAnnotations#c65f6 relayoutBoundary=up11
 I/flutter:         child: RenderExcludeSemantics#ba063 relayoutBoundary=up12
 I/flutter:           child: RenderConstrainedBox#b2d79 relayoutBoundary=up13
 I/flutter:             child: RenderPositionedBox#1f643 NEEDS-LAYOUT
 I/flutter:       child 2: RenderConstrainedBox#09b46 relayoutBoundary=up11
 I/flutter:       child 3: RenderParagraph#8a529 relayoutBoundary=up11 NEEDS-PAINT
 I/flutter:         text: TextSpan
 I/flutter:       child 4: RenderParagraph#c3d22 relayoutBoundary=up11 NEEDS-PAINT
 I/flutter:         text: TextSpan
 I/flutter:       child 5: RenderParagraph#3638a relayoutBoundary=up11 NEEDS-PAINT
 I/flutter:         text: TextSpan
 I/flutter:     subtitle: RenderPadding#c926f relayoutBoundary=up10 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
 I/flutter:       child: RenderPositionedBox#b0950 relayoutBoundary=up11 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
 I/flutter:         child: RenderRepaintBoundary#e3589 relayoutBoundary=up12 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
 I/flutter:           child: RenderCustomPaint#ec130 relayoutBoundary=up13 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
 I/flutter:             child: RenderRepaintBoundary#3fe31 relayoutBoundary=up14 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
 I/flutter:     trailing: RenderConstrainedBox#724c9 relayoutBoundary=up10
 I/flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
```

```
```

```
```

**Comment by vertigeaux** [comment]
Did you ever resolve this issue? I am getting the same error. I haven't quite figured out the exact cause or fix, but I have narrowed it down to a conditional that I am using in my ListTile's subtitle. In my case, there is no problem building the tiles with the conditional item in the subtitle on the initial build. Only when adding or removing an item from the list do I get the error. 

Can you share the source of _buildDeviceAlarmInfoWidget()? Maybe that will help us figure this out.

**Comment by BondarenkoStas** [comment]
@kecson I will leave the reference here for future follow up #40002.

**Comment by toby3000** [comment]
> Did you ever resolve this issue? I am getting the same error. I haven't quite figured out the exact cause or fix, but I have narrowed it down to a conditional that I am using in my ListTile's subtitle. In my case, there is no problem building the tiles with the conditional item in the subtitle on the initial build. Only when adding or removing an item from the list do I get the error.
> 
> Can you share the source of _buildDeviceAlarmInfoWidget()? Maybe that will help us figure this out.

Yes, i have the same problem. Without subtitle there is no problem, but with subtitle i got that strange error.

**Comment by kecson** [comment]
this issue fixed in stable 1.12.13+hotfix.5.
